### ISSUE TITLE
What SKOS concepts should we use to structure the catalog?

### ISSUE THREAD

**Issue by jeff-zucker** [solid/catalog#4]
Coming soon ...

**Comment by jeff-zucker** [comment]
And I think you should not have created  classes for ClassesOfProduct when SKOS concepts exist.  The entire design I created is based on SKOS concepts.  You have consistently ignored or denigrated them without any reasoning.  DCAT and most of the EU Commission SEMIC ontologies use SKOS, so do Gist and DPV and many other important ontologies. Pat McBennett (who has agreed to talk to the three of us when we're ready) is strongly in favor of SKOS and said the design shown in #28 is "excellent".  If you have issues with SKOS please address them in #28.

**Comment by elf-pavlik** [comment]
What about classes which we are re-using from existing vocabs `Person`, `Specification` etc. Did Pat recommend to create separate SKOS concepts? What are the drawbacks of using existing classes as concepts?

**Comment by jeff-zucker** [comment]
> What about classes which we are re-using from existing vocabs Person, Specification etc

There are several ways to combine RDFS and/or OWL classes with SKOS with some subtle differences.  When we have decided on how to replace the ex:namespace with existing namespaces, we can then consider whether we want to co-define the SKOS concepts as being both concepts and classes or subclasses of existing terms.  I think it is premature to begin doing that before we have the namespaces, but it is something we should think about and discuss with Pat. 

I am not sure what you are asking about, but I'm guessing it's me adding  `ex:subType con:ClassOfProduct" to your records which already defined it as a class.  This is not the same as the process above of actually defining the concept as the same thing as the class - it merely says that in addition to being a member of a Class, the instance can be categorized by a  SKOS concept scheme.  The disadvantage of omitting the SKOS concept scheme and only labeling instances by Class, is that we lose the separation illustrated in #28 which decouples the classes and the concepts.  This decoupling allows things of the same class to be assigned to different parts of the concept tree and things in a given part of the concept tree to be members of differing classes.

> What are the drawbacks of using existing classes as concepts?

If you mean defining a SKOS concept as also an RDFS or OWL class, I don't know of any drawbacks and think we will want to do this where possible once we have the namespaces in order.  If you mean using classes *instead of* using SKOS, the disadvantage is that we lose the decoupling mentioned above.

**Comment by jeff-zucker** [comment]
> In that case I would expect them to use classes but probably not catalog specific concept scheme. 

Why not? We are publishing a shape which includes a concept scheme.  Records submitted by others need to conform to that shape.  DCAT catalogs use the predicate `dcat:themeTaxonomy' to point from the catalog profile to a SKOS concept tree which is used to categorize the catalog and that concept tree is integral to the catalog.  The EU commission publishes dozens or hundreds of catalogs which require SKOS concepts as input.  

For records which we already have in the catalog, if someone submits a URL to self-publish and the data at the URL does not match our shape because it does not have whatever we end up calling `ex:subType/dct::type` SKOS concept, we can merge the self-submitted data with the existing record's subType.  For new records that are missing subTypes, we can put those in a to-be-reviewed bin where a human can add the appropriate concepts before the record is published.  

> we would also need core shapes for self publishing and extended shapes for additional information

I do not yet see a need for two shapes - core and extended. What you see as "extended", I see as an integral and necessary part of the catalog.  There may be a need for this in the future but I do not believe it should be used with what I consider a core need for categorizing concepts.  A catalog is a curated and categorized system, not a simple  list.  An un-curated and un-categorized thing is not a catalog, it is a list. A list of thousands of un-categorized items is useful to no one.  

>  catalog will also have small number of original data

What future are you looking at? You're certainly not looking at the present.  Most organizations and many people do not have RDF profiles (currently 92 of 377 Organizations and Persons have WebIDs in the catalog).  Most apps, research papers, tutorials, instructional videos, events, communication channels, etc. do not have any form of RDF profile. I'm not sure we can ever expect them to have their own RDF profiles, certainly not soon. To say that we want to move towards as much self-publishing as possible is a long-term goal that only fits certain kinds of resources and we can not expect it to cover everything or even most things in the catalog in the next few years and in some cases, ever.  The number of records that are not aggregated from elsewhere will never be "small".  

I would encourage the goddess to wave a wand and make all resources have RDF profiles so that we can aggregate them, but I am not holding my breath for her to do so and believe it unwise to design with the expectation that she will. Having a scraper that finds non-RDF data and converts it is a bit unrealistic in my mind.  For example, suppose we scrape the package.json of published packages - the maintainer/author information in the json will be listed as email addresses or github handles, not as WebIDs so we end up with records that do not point to the person's WebID and no way to connect them to that record and thus integrate them with existing catalog data.

**Comment by elf-pavlik** [comment]
> I do not yet see a need for two shapes - core and extended. What you see as "extended", I see as an integral and necessary part of the catalog.

I don't center my view around catalog. Those core shapes could be to some extent incubated in the catalog to become catalog independent. If there is some catalog specific information then I would make sense to me to create catalog specific shapes extending catalog independent shapes.

> What future are you looking at? You're certainly not looking at the present. Most organizations and many people do not have RDF profiles (currently 92 of 377 Organizations and Persons have WebIDs in the catalog). 

If people and organizations don't even have WebID and personal/professional profiles that says a lot about Solid and the lack of adoption. I would expect most of the information I use in solid efforts (subset of full catalog) to be accessible for self publishing relatively soon, especially if people involved actually self publish. Solid efforts for example only shows ~50 people, who might be reasonable candidates for early adopters of Solid.

**Comment by jeff-zucker** [comment]
> If people and organizations don't even have WebID and personal/professional profiles that says a lot about Solid and the lack of adoption.

Right.  And the catalog is meant to promote that adoption.  Yes, hopefully more and more people and organizations will start having profiles and seeing their use in the catalog may spur them on.   However, people and organizations are only part of the problem and getting them all WebIDs will not solve the lack of RDF profiles in other kinds of resources.

**Comment by jeff-zucker** [comment]
The advantage of using existing SKOS Concepts from the specs which name and define and localize to a specific spec the classes of products.  These concepts already exist and have URIs and prefLabels and definitions.  Why not refer to them rather than creating new classes?

**Comment by elf-pavlik** [comment]
I still need to understand best practices around using SKOS and RDFS/OWL together. Are there any problems with any of the following:

* `sopr:Client a skos:Concept, spec:ClassOfProduct .`
* `spec:ClassOfProduct a skos:Concept, rdfs:Class .`

**Comment by elf-pavlik** [comment]
```ttl
# Example: A Company is running 'Direct Marketing' campaigns for
# a purpose (CampaignA), within which it runs another special
# campaign (CampaignB). This may envolve into another campaign C.
# RDFS+SKOS modelling
dpv:Marketing a rdfs:Class, dpv:Purpose .
dpv:DirectMarketing a dpv:Purpose; skos:broader dpv:Marketing .
ex:CampaignA a dpv:Purpose; skos:broader dpv:DirectMarketing .
ex:CampaignB a dpv:Purpose; skos:broader ex:CampaignA .
# valid statements as all objects are instances of dpv:Purpose
ex:Documentation a dpv:Process; dpv:hasPurpose dpv:DirectMarketing .
```

This example seems to use both classes (instances of `rdfs:Class`) and regular instances as SKOS concepts, so at least us doing that shouldn't be a problem. But yeah, let's get more feedback, I will write a bigger example of using RDFS and SKOS together in the catalog.
